sdcm/cluster.py

```diff
@@ -1872,10 +1874,73 @@ def run_nodetool(self, sub_cmd, args="", options="", ignore_status=False):
         """
         cmd = self._gen_nodetool_cmd(sub_cmd, args, options)
         result = self.remoter.run(cmd, ignore_status=ignore_status)
-        self.log.debug("Command '%s' duration -> %s s", result.command,
-                       result.duration)
+        self.log.debug("Command '%s' duration -> %s s" % (result.command, result.duration))
+
         return result
 
+    def check_node_health(self, target_node=None):
+        self.check_nodes_status()
+        self.check_schema_version()
+
+    def check_nodes_status(self, target_node=None):
+        # Validate nodes health
+        # TODO: improve this part (using get_nodetool_status function from ScyllaCluster)
+        ClusterHealthValidatorEvent(type='start', name='NodesStatus', status=True, node=self.name)
+        try:
+            if not target_node:
+                node_type = ''
+            elif target_node and target_node.name == self.name:
+                node_type = 'target '
+            else:
+                node_type = 'regular '
+
+            self.log.info('Status for %snode %s: %s' % (node_type, self.name, self.run_nodetool('status')))
+
+        except Exception as ex:
+            ClusterHealthValidatorEvent(type='end', name='NodesStatus', status=True,
+                                        node=self._private_ip_address,
```

Comment:
what is the reason that you are using private attribute?

Reply:
Missed it in this place (was changed in others). Fixed